packages/component-base/src/gestures.js

```diff
@@ -0,0 +1,1096 @@
+ * @return {void}
+ */
+export function register(recog) {
+  recognizers.push(recog);
+  for (let i = 0; i < recog.emits.length; i++) {
+    gestures[recog.emits[i]] = recog;
+  }
+}
+
+/**
+ * @private
+ * @param {string} evName Event name.
+ * @return {Object} Returns the gesture for the given event name.
+ */
+function _findRecognizerByEvent(evName) {
+  for (let i = 0, r; i < recognizers.length; i++) {
+    r = recognizers[i];
+    for (let j = 0, n; j < r.emits.length; j++) {
+      n = r.emits[j];
+      if (n === evName) {
+        return r;
+      }
+    }
+  }
+  return null;
+}
+
+/**
+ * Sets scrolling direction on node.
+ *
+ * This value is checked on first move, thus it should be called prior to
+ * adding event listeners.
+ *
+ * @param {!EventTarget} node Node to set touch action setting on
+ * @param {string} value Touch action value
+ * @return {void}
+ */
+export function setTouchAction(node, value) {
+  if (HAS_NATIVE_TA && node instanceof HTMLElement) {
+    // NOTE: add touchAction async so that events can be added in
+    // custom element constructors. Otherwise we run afoul of custom
+    // elements restriction against settings attributes (style) in the
+    // constructor.
+    microTask.run(() => {
+      node.style.touchAction = value;
+    });
+  }
+  node[TOUCH_ACTION] = value;
+}
+
+/**
+ * Dispatches an event on the `target` element of `type` with the given
+ * `detail`.
+ * @private
+ * @param {!EventTarget} target The element on which to fire an event.
+ * @param {string} type The type of event to fire.
+ * @param {!Object=} detail The detail object to populate on the event.
+ * @return {void}
+ */
+function _fire(target, type, detail) {
+  let ev = new Event(type, { bubbles: true, cancelable: true, composed: true });
+  ev.detail = detail;
+  wrap(/** @type {!Node} */ (target)).dispatchEvent(ev);
+  // forward `preventDefault` in a clean way
+  if (ev.defaultPrevented) {
+    let preventer = detail.preventer || detail.sourceEvent;
+    if (preventer && preventer.preventDefault) {
+      preventer.preventDefault();
+    }
+  }
+}
+
+/**
+ * Prevents the dispatch and default action of the given event name.
+ *
+ * @param {string} evName Event name.
+ * @return {void}
+ */
+export function prevent(evName) {
+  let recognizer = _findRecognizerByEvent(evName);
+  if (recognizer.info) {
+    recognizer.info.prevent = true;
+  }
+}
+
+/**
+ * Reset the 2500ms timeout on processing mouse input after detecting touch input.
+ *
+ * Touch inputs create synthesized mouse inputs anywhere from 0 to 2000ms after the touch.
+ * This method should only be called during testing with simulated touch inputs.
+ * Calling this method in production may cause duplicate taps or other Gestures.
+ *
+ * @return {void}
+ */
+export function resetMouseCanceller() {
+  if (POINTERSTATE.mouse.mouseIgnoreJob) {
+    POINTERSTATE.mouse.mouseIgnoreJob.flush();
+  }
+}
+
+/* eslint-disable valid-jsdoc */
```

Comment:
Let's consider removing this comment and a corresponding `eslint-enable` comment (we don't use the rule). If we decide to enable it in the future, we will most likely have a lot more ESLint errors elsewhere.

Reply:
Done